### DISCUSSION TITLE
what should i do to make udp working on my private server?

### DISCUSSION THREAD

**Discussion by woshilaiba** [vpnhood/VpnHood#66] [Q&A] [answered]
On my server just two tcp ports are open to public , one is for ssh, another is for  vpnhood.
vpnhood works great on tcp, but not on udp.
I guess that'll be the firewall's fault.
So what ports of udp should i  open to make it work?

**Comment by trudyhood** [comment, marked as answer]
Hi, Thank you for trying VpnHood and reporting the issue,
There are few consideration for using UDP.

1. VpnHood UDP protocol, tunnel UDP over UDP and TCP over TCP, but in non UDP, it tunnels all packets over TCP. So VpnHood uses TCP tunnel for TCP connections even in UDP mode. 

2. We have already detected an issue in recent version that VpnHood doesn't use UDP at all, hopefully a new version will be published today or by start of the next week! :grimacing:

3. VpnHood use a dynamic UDP port, this behavior maybe changed by restricting a UDP port range or share a UDP port for all users. By the way, at this time make sure all UDP ports is open, or temporary disable Firewall to see is it work or not.

4.  You should know that if you are using VpnHood to circumvent Internet Censorship, many aggressive firewall  may completely block all unknown  UDP protocol. In that case you can't use UDP for any VPN.

Let me know if you have any question on your mind.

**Reply by woshilaiba** [reply]
Thank you! I'll wait for the new version.

**Reply by trudyhood** [reply]
You can try version 2.2.283. Just let me know if you encounter any issue.

**Reply by woshilaiba** [reply]
Just the same as before when firewall enabled.
It works like a charm when I disable the firewall.
 Maybe restrict the udp port range which should be set at the server side  is a better  choise.

**Reply by trudyhood** [reply]
For Server: You should open server firewall for any communication port. Is it a network firewall or a firewall installed on server machine?
For Client: Is it a firewall installed on your machine like Windows Firewall or something on your network?

**Reply by woshilaiba** [reply]
server : centos7.9,  firewalld with only udp port 443 and tcp port 443 open for vpnhood.
client  :  oneplus 8t android phone.
UDP mode only works when disable firewalld on server.
the udp port for vpnhood is random, so i cannot specify which udp port to open.
"Maybe restrict the server udp port range which should be set at the server side is a better choise."

**Reply by trudyhood** [reply]
I added an issue for it, but not sure when we implement it.
https://github.com/vpnhood/VpnHood/issues/69
